lib/sqlite.js

```diff
@@ -0,0 +1,51 @@
+'use strict'
+const path = require('path')
+const { open } = require('sqlite')
+const sqlite3 = require('sqlite3')
+if (false) sqlite3.verbose()
+
+const mode = readwrite =>
+  readwrite ?
+    sqlite3.OPEN_READWRITE | sqlite3.OPEN_CREATE :
+    sqlite3.OPEN_READONLY // TODO: check if opening fails
+
+const openWithDriver = driver => (filename, readwrite=false) =>
+  open({
+    filename,
+    mode: mode(readwrite),
+    driver,
+  })
+
+exports.open = openWithDriver(sqlite3.Database)
+exports.openCached = openWithDriver(sqlite3.cached.Database)
+
+exports.withDB = db => proc => db.then(proc)
+
+exports.all = withDB => (...args) => withDB(db => db.all(...args))
+exports.get = withDB => (...args) => withDB(db => db.get(...args))
+exports.run = withDB => (...args) => withDB(db => db.run(...args))
+
+exports.withTx = withDB => proc => withDB(db => new Promise((resolve, reject) =>
+  db.getDatabaseInstance().serialize(() =>
+    db.run("BEGIN TRANSACTION")
+      .then(_ => proc(db))
+      .then(ret => db.run("COMMIT TRANSACTION").then(_ => resolve(ret)))
+      .catch(err => {
+        db.run("ROLLBACK TRANSACTION").then(()=>{},()=>{})
+        reject(err)
+      })
+  )
+))
+
+exports.helpers = dbPromise => {
+  const withDB = exports.withDB(dbPromise)
+  const all = exports.all(withDB)
+  const get = exports.get(withDB)
+  const run = exports.run(withDB)
+  const withTx = exports.withTx(withDB)
+  return { withDB, all, get, run, withTx }
+}
+
+const repeat = (n, x) => Array(n).fill(x)
+exports.makeTuple = n => '(' + repeat(n, '?').join(',') + ')'
+exports.makeValues = (nColumns, nRows) => repeat(nRows, exports.makeTuple(nColumns)).join(',')
```

Comment:
I'm confused with all of the exports, most of them are not really being used anywhere and it makes an awkward read. 

I'm also failing to understand the benefit of everything being 'factory' based when it uses the exact same db object anyways. Why not just do `db.saveCoins` (or whatever queries) on lssettings?

Reply:
`all`/`get`/`run` were used at some point, until I switched to `withTx`.

Reply:
RE factory based: if you mean the `helpers()` it's for usage-convenience.